src/parse/matches/matched_arg.rs

```diff
@@ -125,12 +124,12 @@ impl MatchedArg {
         }
     }
 
-    pub(crate) fn contains_explicit_val(&self, val: Option<&str>) -> bool {
+    pub(crate) fn check_explicit<'a>(&self, predicate: ArgPredicate<'a>) -> bool {
         if self.ty == ValueType::DefaultValue {
             return false;
         }
 
-        if let Some(val) = val {
+        if let ArgPredicate::Equals(val) = predicate {
```

Comment:
Please `match` here
- We get the compiler confirming we are covering each case
- It becomes self-documenting what the `else` branch means